File: implementations/java/src/main/java/com/cae/mapped_exceptions/specifics/NotAuthorizedMappedException.java
package com.cae.mapped_exceptions.specifics;

import com.cae.mapped_exceptions.MappedException;

public class NotAuthorizedMappedException extends MappedException {

    public NotAuthorizedMappedException(String briefPublicMessage, String details) {
        super(briefPublicMessage, details);
    }

    public NotAuthorizedMappedException(String briefPublicMessage) {
        super(briefPublicMessage);
    }

    public NotAuthorizedMappedException(String briefPublicMessage, String details, Exception originalException){
        super(briefPublicMessage, details, originalException);
    }

    public NotAuthorizedMappedException(String briefPublicMessage, Exception originalException){
        super(briefPublicMessage, originalException);
    }

}

File: implementations/python/src/cae_mapped_exceptions/specifics/not_authorized_mapped_exception.py
from cae_mapped_exceptions.mapped_exception import MappedException


class NotAuthorizedMappedException(MappedException):

    def __init__(
            self,
            brief_public_message: str | None = None,
            details: str | None = None,
            original: Exception | None = None
    ):
        super().__init__(brief_public_message, details, original)
